src/zkevm_specs/evm/execution/calldatacopy.py

```diff
@@ -15,11 +15,12 @@ def calldatacopy(instruction: Instruction):
     memory_offset, length = instruction.memory_offset_and_length(memory_offset_word, length_word)
     data_offset = instruction.rlc_to_fq(data_offset_word, N_BYTES_MEMORY_ADDRESS)
 
-    tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId, RW.Read)
     if instruction.curr.is_root:
-        call_data_length = instruction.tx_context_lookup(tx_id, TxContextFieldTag.CallDataLength)
+        src_id = instruction.call_context_lookup(CallContextFieldTag.TxId, RW.Read)
+        call_data_length = instruction.tx_context_lookup(src_id, TxContextFieldTag.CallDataLength)
```

Comment:
The `CallDataLength` in `tx_table` is actually the length of input of transaction, in creation transaction case, the `call_data` is actually empty, so the length of input should not be used. We could read `CallDataLength` from call context which is guaranteed to be length of `call_data`. Note that it adds an extra `rw_table` read, so both markdown and test need an update.

This seems also confusing (my apology)... Perhaps we should rename `CallData*` to `Input*` in `tx_table` to explicitly make them different, or add some more explanation on that.

```suggestion
        call_data_length = instruction.call_context_lookup(CallContextFieldTag.CallDataLength)
```

Reply:
We can do the renaming in a separate PR.

Very good catch, I've fixed this part :)